### PR TITLE
Be more strict with types

### LINKED ISSUE #2888: setExpires/getMaxAge: Notice: A non well formed numeric value encountered
**Guzzle version(s) affected**: 7.3.0
**PHP version**: 7.3.19-1~deb10u1
**cURL version**: 7.66.0

**Description**  
The line [$this->setExpires(\time() + $this->getMaxAge());](https://github.com/guzzle/guzzle/blob/master/src/Cookie/SetCookie.php#L87) generates a "Notice: A non well formed numeric value encountered" when the received Cookie contains an invalid "Max-Age" for the cookie.

**How to reproduce**  
In my case the server returned two comma separated values for Max-Age: "63071999,9843797"

**Possible Solution**  
Either explicitly cast the return value from `$this->getMaxAge()` to "int", or add an additional validation for "digits only".

### PR DESCRIPTION
**Pull request by Nyholm**
Make sure getMaxAge() always returns null or int. Also trigger deprecations if wrong types are passed to SetCookie

This will fix #2888

### CODE REVIEW

**Comment by GrahamCampbell** [comment]
We should probably turn off the authoritative phpdoc types setting in phpstan (or whatever it's called).

**Comment by Nyholm** [comment]
Yeah... But I rather just add this to the baseline. 

Note how I prepare the user and code for version 8.

**Comment by GrahamCampbell** [comment]
Ok, sure. Re-gen the baseline then.

**Comment by Nyholm** [comment]
Baseline is updated and the CI is green.

File: composer.json [hunk 1]
```diff
@@ -32,7 +32,8 @@
         "ext-json": "*",
         "guzzlehttp/promises": "^1.4",
         "guzzlehttp/psr7": "^1.7 || ^2.0",
-        "psr/http-client": "^1.0"
+        "psr/http-client": "^1.0",
+        "symfony/deprecation-contracts": "^2.4"
```
Comment (GrahamCampbell) [inline composer.json:36]: `^2.2` would be better I think. There are no code changes between 2.2.0 and 2.4.0: https://github.com/symfony/deprecation-contracts/compare/v2.2.0...v2.4.0.

**Review by GrahamCampbell — CHANGES_REQUESTED** [review]
One tiny comment. Then good to merge.

File: composer.json (hunk 1 above)
Comment (Nyholm) [reply composer.json]: Good idea!

**Pull request merged**